### DISCUSSION TITLE
File Upload with large filename produces extremely long hash IDs

### DISCUSSION THREAD

**Discussion by akulmehta** [livewire/livewire#7144] [7. Bugs (archived)]
### Livewire version

3.0.10 and 2.12.6

### Which PHP version are you using?

PHP 8.1

### Steps To Reproduce

Livewire file uploads are great but the way the hash ID is generated, it produces extremely long hash IDs for files with long names. While in most cases users would not encounter any errors, if someone is saving the files with the same hash IDs and storing the hash IDs to the database, this can break as sometimes the text can be so long that it does not fit in the DB column. The DB column for a hash is usually preferred to be a string so that the unique() method can make sure that the hash is unique. 

To reproduce this:
1. Setup Livewire File Uploads as described in the docs.
2. Try uploading a file with a really long name e.g. `file_with_a_veryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryvery_long_name.png`
3. Try saving the hashID generated in a regular string column (max length 255).

### Additional information

This issue is caused due how the function `generateHashNameWithOriginalNameEmbedded` works (located in the TemporaryUploadedFile.php). It incorporates the original file name in the meta. As a result, if the filename is very large, it will incorporate the base64 encoding of the original file name into the hash. This greatly increases the hash ID length and can result in the above issue.

Not sure what the long term solution should be in this case and not sure if the meta is used in any other ways but I thought I should report this in case anyone else faces this issue. For the short term obviously I will go ahead and change my DB column type to something more longer.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by lrljoe** [comment]
Do you fancy submitting a failing test PR for this to help the guys get to the bottom of the issue and come up with a solution?

It'd be appreciated by the core team I know!

**Reply by akulmehta** [reply]
@lrljoe I would have loved to do it. But I am a novice and don't know how to generate a failing test for this (not good with testing). I was able to source dive and find the problem source but not sure how to contribute and produce a test. Also unsure if the meta added to the filename is used anywhere else in Livewire.
